openeogeotrellis/deploy/batch_job.py

```diff
@@ -191,8 +194,65 @@ def epsg_code(gps_crs) -> Optional[int]:
+            # New approach: SaveResult has generated metadata already for us
+
+            # Add the projection extension metadata.
+            # When the projection metadata is the same for all assets, then set it at the item level only.
+            # This makes it easier to read, so we see at a glance that all bands have the same proj metadata.
+            projection_metadata = {}
+
+            # We also check whether any asset file was missing or its projection metadata could not be read.
+            # In that case we never write the projection metadata at the item level.
+            missing_files_or_projection_md = False
+
+            for asset_path in asset_metadata.keys():
+                if not Path(asset_path).exists():
+                    # Path should exist so log an error.
+                    # We should only have missing files during unit tests that don't actually use files.
+                    # However, if this happens in production we should see it in the logs.
+                    #
+                    # Also note: As soon as any file is missing we can't really have the same CRS and
+                    # BBox for all assets anymore.
+                    missing_files_or_projection_md = True
+                    logger.error(
+                        "Can not read asset's projection extension metadata, asset file not found: {asset_path}"
+                    )
+                else:
+                    asset_proj_metadata = read_projection_extension_metadata(asset_path)
+                    # If gdal could not extract the projection metadata from the file (The file is corrupt perhaps?).
+                    if asset_proj_metadata:
+                        projection_metadata[asset_path] = asset_proj_metadata
+                    else:
+                        missing_files_or_projection_md = True
+                        logger.error(
+                            "Could not get projection extension metadata for following asset file: {asset_path}"
+                        )
+
+            same_epsg_all_assets = False
+            same_bbox_all_assets = False
+            same_shapes_all_assets = False
+            if not missing_files_or_projection_md:
+                epsgs = {m["proj:epsg"] for m in projection_metadata.values()}
+                same_epsg_all_assets = len(epsgs) == 1
+
+                bboxs = {tuple(m["proj:bbox"]) for m in projection_metadata.values()}
+                same_bbox_all_assets = len(bboxs) == 1
+
+                shapes = {tuple(m["proj:shape"]) for m in projection_metadata.values()}
+                same_shapes_all_assets = len(shapes) == 1
+
+            if same_epsg_all_assets and same_bbox_all_assets and same_shapes_all_assets:
+                # TODO: Should we overwrite existing values for epsg and bbox, or keep what is already there?
+                if not epsg and not metadata.get("epsg"):
+                    epsg = epsgs.pop()
```

Comment:
I'm getting a warning on this line:
> Local variable 'epsgs' might be referenced before assignment

Reply:
I have cleaned up the logic so this could be avoided.
There was an if statement that could be eliminated, and simplifying this made the intention a bit more clear as well.